app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -57,16 +57,17 @@ class AgendaRepositoryImpl(
             }
         }
 
-        emit(
-            reminders.combine(tasks) { reminderFlow, taskFlow ->
-                reminderFlow + taskFlow
-            }.combine(events) { reminderTaskFlow, eventsFlow ->
-                reminderTaskFlow + eventsFlow
+        // Added underscore to get rid of name shadowing error
+        emitAll(
+            reminders.combine(tasks) { _reminders, _tasks ->
+                _reminders + _tasks
+            }.combine(events) { _remindersAndTasks, _events ->
+                _remindersAndTasks + _events
             }.map {
                 it.sortedBy { agendaItem ->
                     agendaItem.startDateAndTime
                 }
-            }.first()
+            }
         )
```

Comment:
The reason that nothing is happening in your flow after this `emitAll()` is that this function will block the flow until all emissions are sent into it. However, since the DB flows never complete (because there could always be new emissions when the table changes), this `emitAll()` will also never complete.
I'd get rid of the extra complexity here and have a simple function that just updates the local DB after getting the agenda items from the network. Then you don't need any flow logic except for just returning the flow from the DB as it is. So something like this:
```
suspend fun updateAgendaItemCache(...) {
    // Retrofit call to get agenda items for the day
    ...
    // Insert all items in local DB
}

// This flow that is returned here will emit new items every time the updateAgendaItemCache function from above is called
fun getAgendaItems(...): Flow<List<AgendaItem>> {
    return combine(dao.getEvents(...), dao.getTasks(...), dao.getReminders(...)) { events, tasks, reminders ->
        (events + tasks + reminders).sortedBy { it.dateAndTime }
    }
}
```